tfjs-backend-webgl/src/gpgpu_math.ts

```diff
@@ -381,14 +381,15 @@ export function makeShaderKey(
       }
       const xRank = x.shape.length;
       const isLogicalShapTexShapeEqual =
-          xRank === 2 && util.arraysEqual(x.shape, xTexShape);
+          uniformShape.length === 2 && util.arraysEqual(x.shape, xTexShape);
       const isScalar = util.sizeFromShape(x.shape) === 1;
       const broadcastDims =
           backend_util.getBroadcastDims(x.shape, output.shape);
       const isInOutTexShapeEqual = !program.packedInputs &&
           xRank === output.shape.length &&
           util.arraysEqual(xTexShape, output.texData.texShape);
-      const isTexShapeGreaterThanOne = program.packedInputs || xRank > 2 ?
+      const isTexShapeGreaterThanOne =
+          program.packedInputs || uniformShape.length > 2 ?
```

Comment:
Some failures are caused by using `xRank` instead of `uniformShape.length`.
Previously, if we have a 3d shape, which maybe squeeze to a 2d shape, we should check whether the 2d shape's texShape is greater than one to be as the part of shaderKey. However, if we use `xRank > 2` as the condition, `isTexShapeGreaterThanOne` will be an empty string which is not we expected.